modules/meta.analysis/R/meta.analysis.summary.R

```diff
@@ -22,74 +22,69 @@
-
-  fail = FALSE
-  sink(file = file.path(outdir,'meta-analysis.log'), append = TRUE, split = TRUE)
-  for (trait in names(mcmc.object)){
+pecan.ma.summary <- function(mcmc.object, pft, outdir, threshold = 1.2, gg = FALSE) {
+  
+  fail <- FALSE
+  sink(file = file.path(outdir, "meta-analysis.log"), append = TRUE, split = TRUE)
+  for (trait in names(mcmc.object)) {
     
-    if(gg){
+    if (gg) {
       gg <- require(ggmcmc)
     }
-    ## new diagnostic plots. But very slow 
-    if(gg){# & !any(grepl('^gg', dir(outdir)))){
-      if(is.mcmc.list(mcmc.object[[trait]])){
+    ## new diagnostic plots. But very slow & !any(grepl('^gg', dir(outdir)))){
+    if (gg) {
+      if (is.mcmc.list(mcmc.object[[trait]])) {
         theme_set(theme_bw())
         ggmcmc(ggs(mcmc.object[[trait]]), 
-               plot = c('ggs_density', 'ggs_traceplot', 'ggs_autocorrelation', 'ggs_Rhat', 'ggs_geweke'),
-               file.path(outdir, paste0("gg.ma.summaryplots.", trait, ".pdf")))        
+               plot = c("ggs_density", "ggs_traceplot", "ggs_autocorrelation", "ggs_Rhat", "ggs_geweke"), 
+               file.path(outdir, paste0("gg.ma.summaryplots.", trait, ".pdf")))
       }
     }
     
     ## reordering maparms so that beta.o etc not sent to end
-    .maparms <- names(mcmc.object[[trait]][1,][1][[1]])
-    .parms <- c('beta.o', 'thetaSD', 'trtSD', 'ySD')
-    maparms <- .maparms[ c(which(.maparms %in% .parms), which(!.maparms %in% .parms))]
-
+    .maparms <- names(mcmc.object[[trait]][1, ][1][[1]])
+    .parms   <- c("beta.o", "thetaSD", "trtSD", "ySD")
+    maparms  <- .maparms[c(which(.maparms %in% .parms), which(!.maparms %in% .parms))]
+    
     ## plots for mcmc diagnosis
-    pdf(file.path(outdir, paste('ma.summaryplots.', trait, '.pdf', sep = '')))
+    pdf(file.path(outdir, paste0("ma.summaryplots.", trait, ".pdf")))
     
     for (i in maparms) {
-      plot(mcmc.object[[trait]][,i], trace = FALSE, density = TRUE,
-           main = paste('summary plots of',i ,'for', pft, trait))
-      box(lwd=2)
-      plot(mcmc.object[[trait]][,i],density = FALSE)
-      box(lwd=2)
-      autocorr.plot(mcmc.object[[trait]][,i][1], xlim = c(1, 50))
-      box(lwd=2)
+      plot(mcmc.object[[trait]][, i], 
+           trace = FALSE,
+           density = TRUE, 
+           main = paste("summary plots of", i, "for", pft, trait))
+      box(lwd = 2)
+      plot(mcmc.object[[trait]][, i], density = FALSE)
+      box(lwd = 2)
+      autocorr.plot(mcmc.object[[trait]][, i][1], xlim = c(1, 50))
+      box(lwd = 2)
     }
     xyplot(mcmc.object[[trait]])
     densityplot(mcmc.object[[trait]])
     acfplot(mcmc.object[[trait]])
     dev.off()
- 
-    ## G-R diagnostics to ensure convergence    
-    gd <- gelman.diag(mcmc.object[[trait]])
-    mpsrf <- round(gd$mpsrf, digits = 3)
+    
+    ## G-R diagnostics to ensure convergence
+    gd            <- gelman.diag(mcmc.object[[trait]])
+    mpsrf         <- round(gd$mpsrf, digits = 3)
```

Comment:
Yep I think I brought this up last time @istfer and you were working on automated convergence diagnostics